### DISCUSSION TITLE
is there any pdf, docx, png preview plugin

### DISCUSSION THREAD

**Discussion by wieniawski-lai** [doublecmd/doublecmd#1246] [General]
I wanna use F4 to preview pdf, docx and png files.
Is there any plugin to support?

**Comment by Skif-off** [comment]
Maybe F3?

OS and version? Application version?

P.S. PNG supports in any case, it's a built-in feature.

**Reply by wieniawski-lai** [reply]
yes, F3 can preview png files.
but pddf, docx is not supported.
My OS is kubuntu2204LTS
doublecmd version is 1.0.11

**Reply by Skif-off** [reply]
DC 1.1.0+ has a new "Office XML (text only)" mode: will show text from XML-based office documents (MS/LO Office, i.e. DOCX/ODT and  XLSX/ODS), without text formatting, but preserving paragraphs.

Also see [list](https://github.com/j2969719/doublecmd-plugins/blob/master/plugins.md#wlxqt5):

DOCX, for example:
`fileinfo_qt`: `fileinfo.sh` contains a command to get text from DOCX files, but you can try to add Pandoc or LibreOffice.
`hx_webkit_qt_crap`: HTML Export from Oracle Outside In Technology allows to save the formatting.
`wlxwebkit_qt_crap`: if you find a suitable utility for converting DOCX to HTML.

PDF: `qtpdfview_qt`, `pdf_crap_qml_qt`, `hx_webkit_qt_crap`.

[Few words](https://github.com/j2969719/doublecmd-plugins/blob/master/dubuild.md) about compilation. If you have problem, try [dc-plugins-for-ubuntu-22.04-*.tar.gz](https://disk.yandex.ru/d/jhgHgkt1ypuhPw/other) (select file and use the "↓" or "↓ Скачать" button).
